guidelines/design-guidelines.md

```diff
@@ -0,0 +1,34 @@
+## Configuration API's
+
+All configuration API's should be code first since that allows us to best evolve them while guiding users with obsolete messages.
```

Comment:
Suggest:

> All configuration API's should be ~code first~ *code-first to* ~since that allows~ *allow* us to ~best~ evolve them while guiding users with ~obsolete~ *deprecation* messages.